<h1>DATA ANALYSIS OF NEW YORK CITI BIKERS IN PYTHON </h1>


<h2>Table of Contents</h2>

<div class="alert alert-block alert-info" style="margin-top: 20px">
<ul>
    <li><a href="https://#remove duplicates">Identify and remove duplicates</a></li>
    <li><a href="https://#identify_handle_missing_values">Identify and handle missing values</a>
        <ul>
            <li><a href="https://#identify_missing_values">Identify missing values</a></li>
            <li><a href="https://#deal_missing_values">Deal with missing values</a></li>
            <li><a href="https://#correct_data_format">Correct data format</a></li>
        </ul>
    </li>
    <li><a href="https://#data_standardization">Data standardization</a></li>
    <li><a href="https://#data_normalization">Data normalization (centering/scaling)</a></li>
    <li><a href="https://#binning">Binning</a></li>
    <li><a href="https://#indicator">Indicator variable</a></li>
</ul>

</div>

<hr>


# Import software libraries

In [1]:
import pandas as pd

# Load the dataset as a `DataFrame`

In [2]:
NY_Citi_Bikes_data=pd.read_csv('Datasets/New York Citi Bikes_Raw Data - NYCitiBikes (1).csv')

In [3]:
NY_Citi_Bikes_data

Start Time       Stop Time  Start Station ID Start Station Name  \
0      01-01-17 00:38    1-1-17 01:03              3194    McGinley Square   
1      01-01-17 01:47  01-01-17 01:58              3183     Exchange Place   
2      01-01-17 01:47  01-01-17 01:58              3183     Exchange Place   
3      01-01-17 01:56  01-01-17 02:00              3186      Grove St PATH   
4        1-1-17 02:12  01-01-17 02:23              3270    Jersey & 6th St   
...               ...             ...               ...                ...   
20395  31-03-17 21:06  31-03-17 21:10              3186      Grove St PATH   
20396  31-03-17 21:13   31-3-17 21:18              3195            Sip Ave   
20397  31-03-17 21:13   31-3-17 21:18              3195            Sip Ave   
20398  31-03-17 21:26  31-03-17 21:30              3195            Sip Ave   
20399   31-3-17 23:20  31-03-17 23:30              3272       Jersey & 3rd   

       End Station ID     End Station Name  Bike ID   User Type  Birth Year  \
0                3271  Danforth Light Rail    24668  Subscriber        1961   
1                3203        Hamilton Park    26167  Subscriber        1993   
2                3203        Hamilton Park    26167  Subscriber        1993   
3                3270      Jersey & 6th St    24604  Subscriber        1970   
4                3206              Hilltop    24641  Subscriber        1978   
...               ...                  ...      ...         ...         ...   
20395            3203        Hamilton Park    24608  Subscriber        1968   
20396            3194      McGinley Square    29219  Subscriber        1956   
20397            3194      McGinley Square    29219  Subscriber        1956   
20398            3201               Dey St    29276  Subscriber        1970   
20399            3268       Lafayette Park    24398  Subscriber        1984   

       Age Age Groups  Trip Duration Trip_Duration_in_min  Month  Season  \
0       60      55-64           1513                   25      1  Winter   
1       28      25-34            639                   11      1  Winter   
2       28      25-34            639                   11      1  Winter   
3       51      45-54            258                    4      1  Winter   
4       43      35-44            663                   11      1  Winter   
...    ...        ...            ...                  ...    ...     ...   
20395   53      45-54            244                    4      3  Spring   
20396   65      65-74            264                    4      3  Spring   
20397   65      65-74            264                    4      3  Spring   
20398   51      45-54            239                    4      3  Spring   
20399   37      35-44            618                   10      3  Spring   

       Temperature Weekday  
0               10  Sunday  
1               10  Sunday  
2               10  Sunday  
3               10  Sunday  
4               10  Sunday  
...            ...     ...  
20395           18  Friday  
20396           18  Friday  
20397           18  Friday  
20398           18  Friday  
20399           18  Friday  

[20400 rows x 17 columns]

In [4]:
NY_Citi_Bikes_data.head(3)

Start Time       Stop Time  Start Station ID Start Station Name  \
0  01-01-17 00:38    1-1-17 01:03              3194    McGinley Square   
1  01-01-17 01:47  01-01-17 01:58              3183     Exchange Place   
2  01-01-17 01:47  01-01-17 01:58              3183     Exchange Place   

   End Station ID     End Station Name  Bike ID   User Type  Birth Year  Age  \
0            3271  Danforth Light Rail    24668  Subscriber        1961   60   
1            3203        Hamilton Park    26167  Subscriber        1993   28   
2            3203        Hamilton Park    26167  Subscriber        1993   28   

  Age Groups  Trip Duration Trip_Duration_in_min  Month  Season  Temperature  \
0      55-64           1513                   25      1  Winter           10   
1      25-34            639                   11      1  Winter           10   
2      25-34            639                   11      1  Winter           10   

  Weekday  
0  Sunday  
1  Sunday  
2  Sunday

# Identify all duplicated data

In [5]:
duplicated_data=NY_Citi_Bikes_data[NY_Citi_Bikes_data.duplicated(keep=False)]

print('number of rows with duplicated data:', 
      duplicated_data.shape[0])

number of rows with duplicated data: 3789


In [6]:
duplicated_data

Start Time       Stop Time  Start Station ID Start Station Name  \
1      01-01-17 01:47  01-01-17 01:58              3183     Exchange Place   
2      01-01-17 01:47  01-01-17 01:58              3183     Exchange Place   
8      01-01-17 03:52  01-01-17 03:55              3203      Hamilton Park   
9      01-01-17 03:52  01-01-17 03:55              3203      Hamilton Park   
20     01-01-17 10:03  01-01-17 10:10              3267       Morris Canal   
...               ...             ...               ...                ...   
20391  31-03-17 20:46  31-03-17 20:54              3186      Grove St PATH   
20394  31-03-17 21:06  31-03-17 21:10              3186      Grove St PATH   
20395  31-03-17 21:06  31-03-17 21:10              3186      Grove St PATH   
20396  31-03-17 21:13   31-3-17 21:18              3195            Sip Ave   
20397  31-03-17 21:13   31-3-17 21:18              3195            Sip Ave   

       End Station ID End Station Name  Bike ID   User Type  Birth Year  Age  \
1                3203    Hamilton Park    26167  Subscriber        1993   28   
2                3203    Hamilton Park    26167  Subscriber        1993   28   
8                3213   Van Vorst Park    24442  Subscriber        1986   35   
9                3213   Van Vorst Park    24442  Subscriber        1986   35   
20               3273     Manila & 1st    26215  Subscriber        1957   64   
...               ...              ...      ...         ...         ...  ...   
20391            3203    Hamilton Park    24558  Subscriber        1967   54   
20394            3203    Hamilton Park    24608  Subscriber        1968   53   
20395            3203    Hamilton Park    24608  Subscriber        1968   53   
20396            3194  McGinley Square    29219  Subscriber        1956   65   
20397            3194  McGinley Square    29219  Subscriber        1956   65   

      Age Groups  Trip Duration Trip_Duration_in_min  Month  Season  \
1          25-34            639                   11      1  Winter   
2          25-34            639                   11      1  Winter   
8          35-44            204                    3      1  Winter   
9          35-44            204                    3      1  Winter   
20         55-64            424                    7      1  Winter   
...          ...            ...                  ...    ...     ...   
20391      45-54            467                    8      3  Spring   
20394      45-54            244                    4      3  Spring   
20395      45-54            244                    4      3  Spring   
20396      65-74            264                    4      3  Spring   
20397      65-74            264                    4      3  Spring   

       Temperature Weekday  
1               10  Sunday  
2               10  Sunday  
8               10  Sunday  
9               10  Sunday  
20              10  Sunday  
...            ...     ...  
20391           18  Friday  
20394           18  Friday  
20395           18  Friday  
20396           18  Friday  
20397           18  Friday  

[3789 rows x 17 columns]

# Remove the duplicated data

In [7]:
NY_Citi_Bikes_data_wt_duplicates=NY_Citi_Bikes_data[~NY_Citi_Bikes_data.duplicated()]

In [8]:
NY_Citi_Bikes_data_wt_duplicates

Start Time       Stop Time  Start Station ID Start Station Name  \
0      01-01-17 00:38    1-1-17 01:03              3194    McGinley Square   
1      01-01-17 01:47  01-01-17 01:58              3183     Exchange Place   
3      01-01-17 01:56  01-01-17 02:00              3186      Grove St PATH   
4        1-1-17 02:12  01-01-17 02:23              3270    Jersey & 6th St   
5      01-01-17 02:22    1-1-17 02:31              3212    Christ Hospital   
...               ...             ...               ...                ...   
20393   31-3-17 21:03   31-3-17 21:12              3278   Monmouth and 6th   
20394  31-03-17 21:06  31-03-17 21:10              3186      Grove St PATH   
20396  31-03-17 21:13   31-3-17 21:18              3195            Sip Ave   
20398  31-03-17 21:26  31-03-17 21:30              3195            Sip Ave   
20399   31-3-17 23:20  31-03-17 23:30              3272       Jersey & 3rd   

       End Station ID       End Station Name  Bike ID   User Type  Birth Year  \
0                3271    Danforth Light Rail    24668  Subscriber        1961   
1                3203          Hamilton Park    26167  Subscriber        1993   
3                3270        Jersey & 6th St    24604  Subscriber        1970   
4                3206                Hilltop    24641  Subscriber        1978   
5                3225  Baldwin at Montgomery    24520  Subscriber        1987   
...               ...                    ...      ...         ...         ...   
20393            3278       Monmouth and 6th    26221  Subscriber        1958   
20394            3203          Hamilton Park    24608  Subscriber        1968   
20396            3194        McGinley Square    29219  Subscriber        1956   
20398            3201                 Dey St    29276  Subscriber        1970   
20399            3268         Lafayette Park    24398  Subscriber        1984   

       Age Age Groups  Trip Duration Trip_Duration_in_min  Month  Season  \
0       60      55-64           1513                   25      1  Winter   
1       28      25-34            639                   11      1  Winter   
3       51      45-54            258                    4      1  Winter   
4       43      35-44            663                   11      1  Winter   
5       34      25-34            535                    9      1  Winter   
...    ...        ...            ...                  ...    ...     ...   
20393   63      55-64            577                   10      3  Spring   
20394   53      45-54            244                    4      3  Spring   
20396   65      65-74            264                    4      3  Spring   
20398   51      45-54            239                    4      3  Spring   
20399   37      35-44            618                   10      3  Spring   

       Temperature Weekday  
0               10  Sunday  
1               10  Sunday  
3               10  Sunday  
4               10  Sunday  
5               10  Sunday  
...            ...     ...  
20393           18  Friday  
20394           18  Friday  
20396           18  Friday  
20398           18  Friday  
20399           18  Friday  

[18450 rows x 17 columns]

# Identify and handle missing values

In the dataset, missing data comes with the question mark "?". We replace "?" with NaN (Not a Number), Python's default missing value marker for reasons of computational speed and convenience. Here we use the function:
.replace(A, B, inplace = True) 
to replace A by B.

In [9]:
import numpy as np

# replace "?" to NaN
NY_Citi_Bikes_data_wt_duplicates.replace("?", np.nan, inplace = True)


C:\Users\kemik\AppData\Roaming\jupyterlab-desktop\jlab_server\lib\site-packages\pandas\core\frame.py:5238: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return super().replace(


# Evaluating for Missing Data

The missing values are converted by default. We use the following functions to identify these missing values. There are two methods to detect missing data:

.isnull()
.notnull()
The output is a boolean value indicating whether the value that is passed into the argument is in fact missing data.

In [10]:
missing_data =NY_Citi_Bikes_data_wt_duplicates.isnull()
missing_data.head(5)

Start Time  Stop Time  Start Station ID  Start Station Name  \
0       False      False             False               False   
1       False      False             False               False   
3       False      False             False               False   
4       False      False             False               False   
5       False      False             False               False   

   End Station ID  End Station Name  Bike ID  User Type  Birth Year    Age  \
0           False             False    False      False       False  False   
1           False             False    False      False       False  False   
3           False             False    False      False       False  False   
4           False             False    False      False       False  False   
5           False             False    False      False       False  False   

   Age Groups  Trip Duration  Trip_Duration_in_min  Month  Season  \
0       False          False                 False  False   False   
1       False          False                 False  False   False   
3       False          False                 False  False   False   
4       False          False                 False  False   False   
5       False          False                 False  False   False   

   Temperature  Weekday  
0        False    False  
1        False    False  
3        False    False  
4        False    False  
5        False    False

"True" means the value is a missing value while "False" means the value is not a missing value.

# Count missing values in each column

Using a for loop in Python, we can quickly figure out the number of missing values in each column. As mentioned above, "True" represents a missing value and "False" means the value is present in the dataset. In the body of the for loop the method ".value_counts()" counts the number of "True" values.

In [11]:
for column in missing_data.columns.values.tolist():
    print(column)
    print (missing_data[column].value_counts())
    print("") 

Start Time
False    18450
Name: Start Time, dtype: int64

Stop Time
False    18450
Name: Stop Time, dtype: int64

Start Station ID
False    18450
Name: Start Station ID, dtype: int64

Start Station Name
False    18450
Name: Start Station Name, dtype: int64

End Station ID
False    18450
Name: End Station ID, dtype: int64

End Station Name
False    18449
True         1
Name: End Station Name, dtype: int64

Bike ID
False    18450
Name: Bike ID, dtype: int64

User Type
False    18450
Name: User Type, dtype: int64

Birth Year
False    18450
Name: Birth Year, dtype: int64

Age
False    18450
Name: Age, dtype: int64

Age Groups
False    18450
Name: Age Groups, dtype: int64

Trip Duration
False    18450
Name: Trip Duration, dtype: int64

Trip_Duration_in_min
False    18450
Name: Trip_Duration_in_min, dtype: int64

Month
False    18450
Name: Month, dtype: int64

Season
False    18450
Name: Season, dtype: int64

Temperature
False    18450
Name: Temperature, dtype: int64

Weekday
False    18450


Based on the summary above, each column has 18450 rows of data and only one of the columns containing missing data:

<ul>
    <li>"End Station Name": 1 missing data</li>
</ul>
To deal with this missing data, we will be deleting the entire row
let's drop all rows that do not have End Station Name data

In [12]:
# Drop whole row with NaN in "End Station Name" column
NY_Citi_Bikes_data_wt_duplicates.dropna(subset=["End Station Name"], axis=0, inplace=True)

# reset index, because we droped one row
NY_Citi_Bikes_data_wt_duplicates.reset_index(drop=True, inplace=True)

C:\Users\kemik\AppData\Roaming\jupyterlab-desktop\jlab_server\lib\site-packages\pandas\util\_decorators.py:311: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return func(*args, **kwargs)


In [13]:
NY_Citi_Bikes_data_wt_duplicates

Start Time       Stop Time  Start Station ID Start Station Name  \
0      01-01-17 00:38    1-1-17 01:03              3194    McGinley Square   
1      01-01-17 01:47  01-01-17 01:58              3183     Exchange Place   
2      01-01-17 01:56  01-01-17 02:00              3186      Grove St PATH   
3        1-1-17 02:12  01-01-17 02:23              3270    Jersey & 6th St   
4      01-01-17 02:22    1-1-17 02:31              3212    Christ Hospital   
...               ...             ...               ...                ...   
18444   31-3-17 21:03   31-3-17 21:12              3278   Monmouth and 6th   
18445  31-03-17 21:06  31-03-17 21:10              3186      Grove St PATH   
18446  31-03-17 21:13   31-3-17 21:18              3195            Sip Ave   
18447  31-03-17 21:26  31-03-17 21:30              3195            Sip Ave   
18448   31-3-17 23:20  31-03-17 23:30              3272       Jersey & 3rd   

       End Station ID       End Station Name  Bike ID   User Type  Birth Year  \
0                3271    Danforth Light Rail    24668  Subscriber        1961   
1                3203          Hamilton Park    26167  Subscriber        1993   
2                3270        Jersey & 6th St    24604  Subscriber        1970   
3                3206                Hilltop    24641  Subscriber        1978   
4                3225  Baldwin at Montgomery    24520  Subscriber        1987   
...               ...                    ...      ...         ...         ...   
18444            3278       Monmouth and 6th    26221  Subscriber        1958   
18445            3203          Hamilton Park    24608  Subscriber        1968   
18446            3194        McGinley Square    29219  Subscriber        1956   
18447            3201                 Dey St    29276  Subscriber        1970   
18448            3268         Lafayette Park    24398  Subscriber        1984   

       Age Age Groups  Trip Duration Trip_Duration_in_min  Month  Season  \
0       60      55-64           1513                   25      1  Winter   
1       28      25-34            639                   11      1  Winter   
2       51      45-54            258                    4      1  Winter   
3       43      35-44            663                   11      1  Winter   
4       34      25-34            535                    9      1  Winter   
...    ...        ...            ...                  ...    ...     ...   
18444   63      55-64            577                   10      3  Spring   
18445   53      45-54            244                    4      3  Spring   
18446   65      65-74            264                    4      3  Spring   
18447   51      45-54            239                    4      3  Spring   
18448   37      35-44            618                   10      3  Spring   

       Temperature Weekday  
0               10  Sunday  
1               10  Sunday  
2               10  Sunday  
3               10  Sunday  
4               10  Sunday  
...            ...     ...  
18444           18  Friday  
18445           18  Friday  
18446           18  Friday  
18447           18  Friday  
18448           18  Friday  

[18449 rows x 17 columns]

Good! Now, we have a dataset with no duplicates and no missing values.

# Check to see if all data formats are correct

In [14]:
NY_Citi_Bikes_data_wt_duplicates.dtypes

Start Time              object
Stop Time               object
Start Station ID         int64
Start Station Name      object
End Station ID           int64
End Station Name        object
Bike ID                  int64
User Type               object
Birth Year               int64
Age                      int64
Age Groups              object
Trip Duration            int64
Trip_Duration_in_min    object
Month                    int64
Season                  object
Temperature              int64
Weekday                 object
dtype: object

In [15]:
NY_Citi_Bikes_data_wt_duplicates.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18449 entries, 0 to 18448
Data columns (total 17 columns):
 #   Column                Non-Null Count  Dtype 
---  ------                --------------  ----- 
 0   Start Time            18449 non-null  object
 1   Stop Time             18449 non-null  object
 2   Start Station ID      18449 non-null  int64 
 3   Start Station Name    18449 non-null  object
 4   End Station ID        18449 non-null  int64 
 5   End Station Name      18449 non-null  object
 6   Bike ID               18449 non-null  int64 
 7   User Type             18449 non-null  object
 8   Birth Year            18449 non-null  int64 
 9   Age                   18449 non-null  int64 
 10  Age Groups            18449 non-null  object
 11  Trip Duration         18449 non-null  int64 
 12  Trip_Duration_in_min  18449 non-null  object
 13  Month                 18449 non-null  int64 
 14  Season                18449 non-null  object
 15  Temperature           18449 non-null

In [16]:
#NY_Citi_Bikes_data_wt_duplicates['Start Time']=pd.to_datetime(NY_Citi_Bikes_data_wt_duplicates['Start Time'], format='%d-%m-%Y %H:%M')

Remove all commas from the column 'Trip_Duration_in_min'

In [17]:
#Remove all commas from the column 
NY_Citi_Bikes_data_wt_duplicates['Trip_Duration_in_min']=NY_Citi_Bikes_data_wt_duplicates['Trip_Duration_in_min'].str.replace(',','')


C:\Users\kemik\AppData\Local\Temp\ipykernel_1160\2263142245.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  NY_Citi_Bikes_data_wt_duplicates['Trip_Duration_in_min']=NY_Citi_Bikes_data_wt_duplicates['Trip_Duration_in_min'].str.replace(',','')


In [18]:
#Change the column data type to int64
NY_Citi_Bikes_data_wt_duplicates['Trip_Duration_in_min']=NY_Citi_Bikes_data_wt_duplicates['Trip_Duration_in_min'].astype('int64')

C:\Users\kemik\AppData\Local\Temp\ipykernel_1160\2480820339.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  NY_Citi_Bikes_data_wt_duplicates['Trip_Duration_in_min']=NY_Citi_Bikes_data_wt_duplicates['Trip_Duration_in_min'].astype('int64')


In [19]:
NY_Citi_Bikes_data_wt_duplicates.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18449 entries, 0 to 18448
Data columns (total 17 columns):
 #   Column                Non-Null Count  Dtype 
---  ------                --------------  ----- 
 0   Start Time            18449 non-null  object
 1   Stop Time             18449 non-null  object
 2   Start Station ID      18449 non-null  int64 
 3   Start Station Name    18449 non-null  object
 4   End Station ID        18449 non-null  int64 
 5   End Station Name      18449 non-null  object
 6   Bike ID               18449 non-null  int64 
 7   User Type             18449 non-null  object
 8   Birth Year            18449 non-null  int64 
 9   Age                   18449 non-null  int64 
 10  Age Groups            18449 non-null  object
 11  Trip Duration         18449 non-null  int64 
 12  Trip_Duration_in_min  18449 non-null  int64 
 13  Month                 18449 non-null  int64 
 14  Season                18449 non-null  object
 15  Temperature           18449 non-null

In [20]:
NY_Citi_Bikes_data_wt_duplicates

Start Time       Stop Time  Start Station ID Start Station Name  \
0      01-01-17 00:38    1-1-17 01:03              3194    McGinley Square   
1      01-01-17 01:47  01-01-17 01:58              3183     Exchange Place   
2      01-01-17 01:56  01-01-17 02:00              3186      Grove St PATH   
3        1-1-17 02:12  01-01-17 02:23              3270    Jersey & 6th St   
4      01-01-17 02:22    1-1-17 02:31              3212    Christ Hospital   
...               ...             ...               ...                ...   
18444   31-3-17 21:03   31-3-17 21:12              3278   Monmouth and 6th   
18445  31-03-17 21:06  31-03-17 21:10              3186      Grove St PATH   
18446  31-03-17 21:13   31-3-17 21:18              3195            Sip Ave   
18447  31-03-17 21:26  31-03-17 21:30              3195            Sip Ave   
18448   31-3-17 23:20  31-03-17 23:30              3272       Jersey & 3rd   

       End Station ID       End Station Name  Bike ID   User Type  Birth Year  \
0                3271    Danforth Light Rail    24668  Subscriber        1961   
1                3203          Hamilton Park    26167  Subscriber        1993   
2                3270        Jersey & 6th St    24604  Subscriber        1970   
3                3206                Hilltop    24641  Subscriber        1978   
4                3225  Baldwin at Montgomery    24520  Subscriber        1987   
...               ...                    ...      ...         ...         ...   
18444            3278       Monmouth and 6th    26221  Subscriber        1958   
18445            3203          Hamilton Park    24608  Subscriber        1968   
18446            3194        McGinley Square    29219  Subscriber        1956   
18447            3201                 Dey St    29276  Subscriber        1970   
18448            3268         Lafayette Park    24398  Subscriber        1984   

       Age Age Groups  Trip Duration  Trip_Duration_in_min  Month  Season  \
0       60      55-64           1513                    25      1  Winter   
1       28      25-34            639                    11      1  Winter   
2       51      45-54            258                     4      1  Winter   
3       43      35-44            663                    11      1  Winter   
4       34      25-34            535                     9      1  Winter   
...    ...        ...            ...                   ...    ...     ...   
18444   63      55-64            577                    10      3  Spring   
18445   53      45-54            244                     4      3  Spring   
18446   65      65-74            264                     4      3  Spring   
18447   51      45-54            239                     4      3  Spring   
18448   37      35-44            618                    10      3  Spring   

       Temperature Weekday  
0               10  Sunday  
1               10  Sunday  
2               10  Sunday  
3               10  Sunday  
4               10  Sunday  
...            ...     ...  
18444           18  Friday  
18445           18  Friday  
18446           18  Friday  
18447           18  Friday  
18448           18  Friday  

[18449 rows x 17 columns]

# Calculate descriptive statistics for “Trip duration” Column

In [31]:
NY_Citi_Bikes_data_wt_duplicates['Trip_Duration_in_min'].mean()

9.58686107648111

In [32]:
NY_Citi_Bikes_data_wt_duplicates['Trip_Duration_in_min'].median()

5.0

In [33]:
NY_Citi_Bikes_data_wt_duplicates['Trip_Duration_in_min'].min()

1

In [34]:
NY_Citi_Bikes_data_wt_duplicates['Trip_Duration_in_min'].max()

6515

# Calculate descriptive statistics for “Age” Column

In [35]:
NY_Citi_Bikes_data_wt_duplicates['Age'].mean()

41.70545829042224

In [36]:
NY_Citi_Bikes_data_wt_duplicates['Age'].median()

39.0

In [37]:
NY_Citi_Bikes_data_wt_duplicates['Age'].min()

22

In [38]:
NY_Citi_Bikes_data_wt_duplicates['Age'].max()

90

In [27]:
NY_Citi_Bikes_data_wt_duplicates.describe()

Start Station ID  End Station ID       Bike ID    Birth Year  \
count      18449.000000    18449.000000  18449.000000  18449.000000   
mean        3215.832186     3211.702911  25299.826603   1979.294542   
std           34.556509       77.179994    978.647165     10.084996   
min         3183.000000      152.000000  15084.000000   1931.000000   
25%         3186.000000     3186.000000  24524.000000   1974.000000   
50%         3203.000000     3202.000000  24679.000000   1982.000000   
75%         3267.000000     3217.000000  26220.000000   1986.000000   
max         3281.000000     3442.000000  29296.000000   1999.000000   

                Age  Trip Duration  Trip_Duration_in_min         Month  \
count  18449.000000   18449.000000          18449.000000  18449.000000   
mean      41.705458     575.226245              9.586861      2.185051   
std       10.084996    4211.040109             70.183325      0.823753   
min       22.000000      61.000000              1.000000      1.000000   
25%       35.000000     220.000000              4.000000      1.000000   
50%       39.000000     311.000000              5.000000      2.000000   
75%       47.000000     516.000000              9.000000      3.000000   
max       90.000000  390893.000000           6515.000000      3.000000   

        Temperature  
count  18449.000000  
mean      14.825953  
std        2.395316  
min        9.000000  
25%       13.000000  
50%       15.000000  
75%       17.000000  
max       19.000000